### PR TITLE
Add Mixxx 2.3.0 Release Ebuild

### PR DESCRIPTION
**Pull request by geraldog**
Hello gentoo,

As requested by fordfrog I'm pushing my PR which is an attempt at giving a version bump for Mixxx 2.3.0 Release. 

Please test and let me know how we can improve it to hopefully get Mixxx 2.3.0 distributed through gentoo.

This fixes https://bugs.gentoo.org/show_bug.cgi?id=809305 https://bugs.gentoo.org/show_bug.cgi?id=780888 and https://bugs.gentoo.org/show_bug.cgi?id=797235

### CODE REVIEW

**Comment by geraldog** [comment]
@fordfrog Mixxx 2.3.1 has been released :)

https://mixxx.org/news/2021-09-30-mixxx-2-3-1-released/

In my humble opinion I should skip this one and file a proper Mixxx 2.3.1 PR?

Please advise! Thanks!

**Comment by thesamesam** [comment]
Hi, thanks for this!

Yeah, I'd suggest doing a 'proper' PR (just force push in this one for 2.3.1).

Also, use repoman or pkgdev to make the commits to ensure you have the right commit message format (and the GCO sign off included).

File: media-sound/mixxx/mixxx-2.3.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig
+	dev-qt/qttest:5
+	dev-qt/qtxmlpatterns:5"
+
+PATCHES=(
+	"${FILESDIR}"/mixxx-2.3.0-docs-and-build-type.patch
+	"${FILESDIR}"/mixxx-2.3.0-local-manuals.patch
+	)
```
Comment (thesamesam) [inline media-sound/mixxx/mixxx-2.3.0.ebuild:81]:
```suggestion
)
```

File: media-sound/mixxx/mixxx-2.3.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	mp3? ( media-libs/libmad )
+	mp4? ( media-libs/libmp4v2:= )
+	opus? (	media-libs/opusfile )
+	qtkeychain? ( dev-libs/qtkeychain )
+	shout? ( >=media-libs/libshout-2.4.5 )
+	wavpack? ( media-sound/wavpack )
+	"
+
+DEPEND="${RDEPEND}"
+BDEPEND="virtual/pkgconfig
```
Comment (thesamesam) [inline media-sound/mixxx/mixxx-2.3.0.ebuild:74]: Put virtual/pkgconfig last

File: media-sound/mixxx/mixxx-2.3.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+		-DINSTALL_USER_UDEV_RULES=OFF
+		-DWAVPACK="$(usex wavpack on off)"
+		-DQTKEYCHAIN="$(usex qtkeychain on off)"
+		-DKEYFINDER="$(usex keyfinder on off)"
+	)
+
+	if [[ "${PV}" == 9999 ]] ; then
+	local mycmakeargs+=(
+		-DENGINEPRIME="OFF"
+
```
Comment (thesamesam) [inline media-sound/mixxx/mixxx-2.3.0.ebuild:109]:
```suggestion
```

File: media-sound/mixxx/mixxx-2.3.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	"${FILESDIR}"/mixxx-2.3.0-docs-and-build-type.patch
+	"${FILESDIR}"/mixxx-2.3.0-local-manuals.patch
+	)
+
+src_prepare() {
+	cmake_src_prepare
+}
+
+src_configure() {
+
```
Comment (thesamesam) [inline media-sound/mixxx/mixxx-2.3.0.ebuild:88]:
```suggestion
```

File: media-sound/mixxx/mixxx-2.3.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+		-DHID="$(usex hid on off)"
+		-DLILV="$(usex lv2 on off)"
+		-DMAD="$(usex mp3 on off)"
+		-DOPTIMIZE="off"
+		-DCCACHE_SUPPORT="off"
+		-DOPUS="$(usex opus on off)"
+		-DBROADCAST="$(usex shout on off)"
+		-DVINYLCONTROL="on"
+		-DINSTALL_USER_UDEV_RULES=OFF
+		-DWAVPACK="$(usex wavpack on off)"
```
Comment (thesamesam) [inline media-sound/mixxx/mixxx-2.3.0.ebuild:101]: I'd not bother with the quotes on any of these, personally

File: media-sound/mixxx/mixxx-2.3.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+	if [[ "${PV}" == 9999 ]] ; then
+	local mycmakeargs+=(
+		-DENGINEPRIME="OFF"
+
+	)
+	fi
+	cmake_src_configure
+}
+
+src_compile() {
```
Comment (thesamesam) [inline media-sound/mixxx/mixxx-2.3.0.ebuild:115]: This should Just Work if you delete the explicit phase definition

**Review by thesamesam — COMMENTED** [review]
Ebuild looks pretty good to me as a non-sound expert. Commented some style nits.

**Comment by geraldog** [comment]
Thank you @thesamesam for your tips and your style nit-picking!

I'll close this one, since I consider Mixxx 2.3.0 already obsoleted by bugfix release 2.3.1.

I'll look into producing the Mixxx 2.3.1 ASAP

**Pull request closed without merging**